### PR TITLE
feat: update redis cache inline during changelog computation

### PR DESCRIPTION
**Pull request by capJavert**
Fix edge case where if redis value is empty boot `changelog` will always be `false`.

### CODE REVIEW

File: src/routes/boot.ts [hunk 1]
```diff
@@ -160,6 +161,34 @@ const setAuthCookie = async (
...
+          },
+        },
+      });
+
+    if (post) {
+      lastChangelogFromRedis = post.createdAt.toISOString();
+    }
+  }
+
+  await setRedisObject(REDIS_CHANGELOG_KEY, lastChangelogFromRedis);
```
Comment (idoshamun) [inline src/routes/boot.ts:187]: This should go inside the `if` so we will not set it constantly

File: __tests__/boot.ts [hunk 2]
```diff
@@ -282,6 +282,74 @@ describe('boot alerts', () => {
...
+    alerts['lastChangelog'] = '2023-02-05T12:00:00.000Z';
+    alerts['changelog'] = true;
+    delete alerts['userId'];
+    const res = await request(app.server)
+      .get(BASE_PATH)
+      .set('Cookie', 'ory_kratos_session=value;')
+      .expect(200);
+    expect(res.body.alerts).toEqual(alerts);
+    expect(await getRedisObject(REDIS_CHANGELOG_KEY)).toEqual(
+      post?.createdAt.toISOString(),
```
Comment (idoshamun) [inline __tests__/boot.ts:316]:
```suggestion
      post.createdAt.toISOString(),
```

File: __tests__/boot.ts [hunk 3]
```diff
@@ -282,6 +282,74 @@ describe('boot alerts', () => {
...
+    alerts['lastChangelog'] = '2023-02-06T12:00:00.000Z';
+    alerts['changelog'] = false;
+    delete alerts['userId'];
+    const res = await request(app.server)
+      .get(BASE_PATH)
+      .set('Cookie', 'ory_kratos_session=value;')
+      .expect(200);
+    expect(res.body.alerts).toEqual(alerts);
+    expect(await getRedisObject(REDIS_CHANGELOG_KEY)).toEqual(
+      post?.createdAt.toISOString(),
```
Comment (idoshamun) [inline __tests__/boot.ts:350]:
```suggestion
      post.createdAt.toISOString(),
```

File: src/routes/boot.ts (hunk 1 above)
Comment (capJavert) [reply src/routes/boot.ts:187]: Good catch 👌

**Review by idoshamun — APPROVED** [review]

**Review by rebelchris — APPROVED** [review]
Nice, thanks for this 🙏

**Pull request merged**